interfaces/tests/file_int/file_int.c

```diff
@@ -0,0 +1,559 @@
+
+#include "file_int_helpers.h"
+
+
+TEST_DEFINE_ENUM_TYPE(FILE_WRITE_ASYNC_RESULT, FILE_WRITE_ASYNC_RESULT)
+TEST_DEFINE_ENUM_TYPE(FILE_READ_ASYNC_RESULT, FILE_READ_ASYNC_RESULT)
+
+typedef struct WRITE_COMPLETE_CONTEXT_TAG
+{
+    int32_t pre_callback_value;
+    volatile_atomic int32_t value;
+    int32_t post_callback_value;
+    bool did_write_succeed;
+}WRITE_COMPLETE_CONTEXT;
+
+typedef struct READ_COMPLETE_CONTEXT_TAG
+{
+    int32_t pre_callback_value;
+    volatile_atomic int32_t value;
+    int32_t post_callback_value;
+    bool did_read_succeed;
+}READ_COMPLETE_CONTEXT;
+
+static void write_callback(void* context, bool is_successful)
+{
+    WRITE_COMPLETE_CONTEXT* write_context = (WRITE_COMPLETE_CONTEXT*)context;
+    ASSERT_ARE_EQUAL(int32_t, write_context->pre_callback_value, interlocked_add(&write_context->value, 0));
+    write_context->did_write_succeed = is_successful;
+    interlocked_exchange(&write_context->value, write_context->post_callback_value);
+    wake_by_address_single(&write_context->value);
+}
+
+static void read_callback(void* context, bool is_successful)
+{
+    READ_COMPLETE_CONTEXT* read_context = (READ_COMPLETE_CONTEXT*)context;
+    ASSERT_ARE_EQUAL(int32_t, read_context->pre_callback_value, interlocked_add(&read_context->value, 0));
+    read_context->did_read_succeed = is_successful;
+    interlocked_exchange(&read_context->value, read_context->post_callback_value);
+    wake_by_address_single(&read_context->value);
+}
+
+static void wait_on_address_helper(volatile_atomic int32_t* address, int32_t* old_value, uint32_t timeout)
+{
+    int32_t current_value;
+    do
+    {
+        wait_on_address(address, old_value, timeout);
+        current_value = interlocked_add(address, 0);
+    }while(current_value == *old_value);
+}
+
+BEGIN_TEST_SUITE(file_int)
+
+
+TEST_SUITE_INITIALIZE(a)
+{
+    g_testByTest = TEST_MUTEX_CREATE();
+    ASSERT_IS_NOT_NULL(g_testByTest);
+}
+
+TEST_SUITE_CLEANUP(b)
+{
+
+    TEST_MUTEX_DESTROY(g_testByTest);
+}
+
+TEST_FUNCTION_INITIALIZE(c)
+{
+    if (TEST_MUTEX_ACQUIRE(g_testByTest))
+    {
+        ASSERT_FAIL("our mutex is ABANDONED. Failure in test framework");
+    }
+
+    delete_all_txt_files();
+}
+
+TEST_FUNCTION_CLEANUP(d)
+{
+    TEST_MUTEX_RELEASE(g_testByTest);
+}
+
+/*Tests_SRS_FILE_43_003: [If a file with name full_file_name does not exist, file_create shall create a file with that name.]*/
+/*Tests_SRS_FILE_43_001: [file_create shall open the file named full_file_name for asynchronous operations and return its handle.]*/
+/*Tests_SRS_FILE_43_006: [file_destroy shall wait for all pending I / O operations to complete.]*/
+/*Tests_SRS_FILE_43_007: [file_destroy shall close the file handle handle.]*/
+TEST_FUNCTION(file_create_creates_new_file)
+{
+    ///arrange
+    EXECUTION_ENGINE_HANDLE execution_engine = execution_engine_create(NULL);
+    char filename[] = "file_create_creates_new_file.txt";
+    ASSERT_IS_FALSE(check_file_exists(filename));

```

Comment:
>check_file_exists [](start = 20, length = 17)

I think we should delete the file in the beginning (make sure the test is starting at a clean slate), and then delete it at the end (tests should try to clean up files they would leave behind, best effort of course because it won't happen if the test fails)